Tests/DiligentToolsTest/src/RenderStateNotationParser/RenderStateNotationParserTest.cpp

```diff
@@ -283,6 +283,257 @@ TEST(Tools_RenderStateNotationParser, InlinePipelineStatesTest)
     }
 }
 
+TEST(Tools_RenderStateNotationParser, ImplicitPipelineStatesTest)
+{
+    RefCntAutoPtr<IRenderStateNotationParser> pParser = LoadFromFile("ImplicitPipelineStates.json");
+    ASSERT_NE(pParser, nullptr);
+
+    {
+        GraphicsPipelineNotation PipelineReference{};
+        PipelineReference.PSODesc.PipelineType = PIPELINE_TYPE_GRAPHICS;
+        PipelineReference.PSODesc.Name         = "Graphics-TestName";
+        PipelineReference.pVSName              = "Shader0-VS";
+        PipelineReference.pPSName              = "Shader0-PS";
+
+        auto pPipeline = pParser->GetGraphicsPipelineStateByName(PipelineReference.PSODesc.Name);
+        ASSERT_NE(pPipeline, nullptr);
+        ASSERT_EQ(*pPipeline, PipelineReference);
```

Comment:
There are ASSERT_ and EXPECT_ macros in gtest.
ASSERT_ is for critical checks (e.g. null pointers.) They abort the test.
EXPECT_ is for non-critical checks (e.g. value does not match expected). They do not abort the test and it continues running